### PR TITLE
draft: create more constructors to use base Entity constructors

### PR DESCRIPTION
**Pull request by maeg02**
DRAFT ONLY, to start the discussion of how to solve #204

Could there be an AppSetting to control this?
Now it's only if(true)

Is there a smarter way to know if the constructors are already there than based on Length <= 1?

### CODE REVIEW

**Comment by maeg02** [comment]
Yes, however it was way past my bedtime and I still hacked away on it, so quality is guaranteed, I will make a manual test, to see if the overloads work as intended with AltKey. You know, more than *it compiles* let shit this... (as is the current state)

File: DLaB.CrmSvcUtilExtensions/Entity/EntityConstructorsGenerator.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+using System.Threading.Tasks;
+
+namespace DLaB.CrmSvcUtilExtensions.Entity
+{
+    internal class EntityConstructorsGenerator : ICustomizeCodeDomService
+    {
+        public void CustomizeCodeDom(CodeCompileUnit codeUnit, IServiceProvider services)
+        {
+            var metadataService = (IMetadataProviderService)services.GetService(typeof(IMetadataProviderService));
+            var metadata = metadataService.LoadMetadata();
```
Comment (daryllabar) [inline DLaB.CrmSvcUtilExtensions/Entity/EntityConstructorsGenerator.cs]: Looks like this isn't used?

File: DLaB.CrmSvcUtilExtensions/Entity/CustomizeCodeDomService.cs [hunk 2]
```diff
@@ -32,6 +32,10 @@ public void CustomizeCodeDom(CodeCompileUnit codeUnit, IServiceProvider services
             {
                 new CodeCustomization(Parameters).CustomizeCodeDom(codeUnit, services);
             }
+            if(true)
```
Comment (daryllabar) [inline DLaB.CrmSvcUtilExtensions/Entity/CustomizeCodeDomService.cs]: So the way this works, is you set a flag in the config that its read as a property.  Then I do a search for the app.Config key of a similar value, and find where it all is updated, and create the new property in the UI/Settings of the actual EBG

Comment (maeg02) [reply DLaB.CrmSvcUtilExtensions/Entity/CustomizeCodeDomService.cs]: yeah, sounds like a plan, if you want to do that, that would be awesome!
Naming things is hard, so if you have something in mind, then that would be good.
Would it be good if the new config would default to false?

File: DLaB.CrmSvcUtilExtensions/Entity/EntityConstructorsGenerator.cs (hunk 1 above)
Comment (maeg02) [reply DLaB.CrmSvcUtilExtensions/Entity/EntityConstructorsGenerator.cs]: Ahm... true. Sharp eyes!
Should I remove it?

Comment (daryllabar) [reply DLaB.CrmSvcUtilExtensions/Entity/EntityConstructorsGenerator.cs]: Yes Please

File: DLaB.CrmSvcUtilExtensions/Entity/CustomizeCodeDomService.cs (hunk 2 above)
Comment (daryllabar) [reply DLaB.CrmSvcUtilExtensions/Entity/CustomizeCodeDomService.cs]: I can do it.  When is just the question...

As far as naming things, GenerateConstructorsSansLogicalName?

**Comment by maeg02** [comment]
An update with the new config, ( a bunch of files now changed)
And update with the actual code:
Moved it earlier in the generated class, and removed unused code

**Pull request merged**